src/agent/debugger/src/dbghelp.rs

```diff
@@ -670,6 +674,94 @@ impl DebugHelpGuard {
             displacement: 0,
         })
     }
+
+    /// Look for a filesystem path to a PDB file using the symbol handler's
+    /// current search path.
+    ///
+    /// This method is effectively a specialization of `SymFindFileInPathW`.
+    ///
+    /// Note: `file_name` may be a full path, but only the file name is used.
+    pub fn sym_find_pdb_file_in_path(
+        &self,
+        process_handle: HANDLE,
+        file_name: impl AsRef<Path>,
+        pdb_signature: u32,
+        pdb_age: u32,
+    ) -> Result<PathBuf> {
+        let file_name = win_util::string::to_wstring(file_name);
+
+        // Must be at least `MAX_PATH` characters in length.
+        let mut found_file_data = Box::new([0u16; MAX_PATH]);
+
+        // Inherit search path used in `SymInitializeW()`. When that is also set
+        // to `NULL`, the default search path is used.
+        let search_path = std::ptr::null_mut();
+
+        // See: https://docs.microsoft.com/en-us/windows/win32/api/dbghelp/nf-dbghelp-symfindfileinpathw#remarks
+        let id = pdb_signature as *mut _;
+        let two = pdb_age;
+        let three = 0;
+
+        // SSRVOPT_DWORD
+        let flags = 0x0002;
+
+        check_winapi(|| unsafe {
+            SymFindFileInPathW(
+                process_handle,
+                search_path,
+                file_name.as_ptr(),
+                id,
+                two,
+                three,
+                flags,
+                found_file_data.as_mut_ptr(),
+                None,
+                std::ptr::null_mut(),
+            )
+        })?;
+
+        // Safety: `found_file_data` must contain at least one NUL byte.
+        //
+        // We zero-initialize `found_file_data`, and assume that `SymFindFileInPathW`
+        // only succeeds if it wrote a NUL-terminated wide string.
+        let found_file =
+            unsafe { win_util::string::os_string_from_wide_ptr(found_file_data.as_ptr()) };
+
+        Ok(found_file.into())
+    }
+
+    pub fn sym_get_search_path(&self, process_handle: HANDLE) -> Result<OsString> {
+        let mut search_path_data = Vec::<u16>::with_capacity(MAX_PATH * 8);
```

Comment:
This was arbitrary; any ideas for a more motivated choice?

Will move this value to a constant and statically ensure it fits in a `u32`.